src/game/plugin/pluginmanager.cpp

```diff
@@ -21,45 +21,269 @@
 #include "Common/StringUtils.h"
 #include "Common/Filesystem.h"
 #include "Common/Logger.h"
+#include "Common/Json/JsonHelper.h"
+#include <stdlib.h>
 
-PluginManager::PluginManager() {
-    infostream << "Start to load plugins...";
-    size_t counter = 0;
-    const filesystem::path path = "./plugins/";
-    if (filesystem::exists(path)) {
-        for (auto&& file : filesystem::directory_iterator(path)) {
-            auto suffix = file.path().extension().string();
-            strToLower(suffix);
-            if (suffix == LibSuffix) {
-                debugstream << "Loading:" << file.path().string();
-                if (loadPlugin(file.path().string()))
-                    counter++;
-            }
-        };
-    }
-    infostream << counter << " plugin(s) loaded";
+struct PluginPair {
```

Comment:
PluginPair doesn't naturally make sense. Please either change to a better name or add comments to explain it.